apis/hive/v1/hiveconfig_types.go

```diff
@@ -529,6 +529,16 @@ type ControllerConfig struct {
 	// This is ONLY for controllers that have been split out into their own pods.
 	// This is ignored for all others.
 	Replicas *int32 `json:"replicas,omitempty"`
+	// MemoryRequest specifies the memory request for the container that the controller is living in.
+	// The format for this string is the same as that specified in the container spec memory requests (e.g. "512Mi").
+	// This is ONLY for controllers that have been split out into their own pods.
+	// This is ignored for all others.
+	MemoryRequest *string `json:"memoryRequest,omitempty"`
+	// MemoryLimit specifies the memory limit for the container that the controller is living in.
+	// The format for this string is the same as that specified in the container spec memory limits (e.g. "512Mi").
+	// This is ONLY for controllers that have been split out into their own pods.
+	// This is ignored for all others.
+	MemoryLimit *string `json:"memoryLimit,omitempty"`
```

Comment:
Rather than bespoke fields that are limited to memory and also need to be parsed with logic duplicated from upstream, can we instead just make a `Resources` field of type corev1.ResourceRequirements, just like what's [defined](https://github.com/openshift/hive/blob/e21e524f15a2334c06aea4f4221c47e2a612cc78/vendor/k8s.io/api/core/v1/types.go#L2503) on Container itself? Then we can just copy the whole thing into the container spec and be done. Also makes it more natural-feeling for the user.